src/Catalog/Monitoring/Validation/Test/EndpointToTest.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test
+{
+    /// <summary>
+    /// Represents an endpoint to run tests against.
+    /// </summary>
+    public abstract class EndpointToTest : EndpointValidationTestRunner
+    {
+        public EndpointToTest(ReadCursor cursor, ValidationTestRunnerFactory factory, ILoggerFactory loggerFactory)
+            : base(factory, loggerFactory)
+        {
+            Cursor = cursor;
+        }
+
+        /// <summary>
+        /// Entries in the catalog past this cursor should not be tested because the endpoint does not contain them.
+        /// </summary>
+        public ReadCursor Cursor { get; private set; }
+
+        /// <summary>
+        /// Constructs all <see cref="Type"/>s from this assembly that are assignable to <see cref="IValidationTestRunner{GetType()}"/> and returns them.
+        /// </summary>
+        protected override IEnumerable<IValidationTestRunner> GetValidationTestRunners(ValidationTestRunnerFactory factory, ILogger logger)
+        {
+            var validationTypes = Assembly.GetExecutingAssembly().GetTypes()
```

Comment:
This code automatically finds any `Type`s that implement `IValidationTestRunner<RegistrationEndpoint>` or `IValidationTestRunner<FlatContainerEndpoint>` (depending on the subclass) and constructs them using a factory. I think the justification is very strong--additional validations that are created are picked up automatically by this code without the need to configure them here.